projects/components/src/search-box/search-box.component.ts

```diff
@@ -89,9 +106,25 @@ export class SearchBoxComponent implements OnInit, OnChanges {
   private setDebouncedSubscription(): void {
     this.subscriptionLifecycle.unsubscribe();
     this.subscriptionLifecycle.add(
-      this.debouncedValueSubject
-        .pipe(debounceTime(this.debounceTime ?? 0))
-        .subscribe(value => this.valueChange.emit(value))
+      combineLatest([this.debouncedValueSubject, this.getDebounceTime()])
+        .pipe(debounce(([_, debounceTime]) => timer(debounceTime)))
+        .subscribe(([value, _]) => this.valueChange.emit(value))
+    );
+  }
+
+  private getDebounceTime(): Observable<number> {
+    return this.enableSearchOnTrigger$.pipe(
+      map(enabled => {
+        const defaultDebounceTime = this.debounceTime ?? 0;
+        // If incremental search mode is enabled, then use the inputs to compute debounce
+        if (this.searchMode === SearchBoxEmitMode.Incremental) {
+          return defaultDebounceTime;
+        }
+
+        // If on-submit search mode is enabled via the FF and the input, then use the overridden debounce time.
+        // Doing 'and' here to ensure the default behaviour continues until overridden by an FF.
+        return enabled && this.searchMode === SearchBoxEmitMode.OnSubmit ? 5000 : defaultDebounceTime;
```

Comment:
nit - enabled here reads like whether the component is enabled, when it's more specific than that.

Reply:
updated